src/paho/mqtt/publish.py

```diff
@@ -23,49 +23,45 @@
 import paho.mqtt as mqtt
 
 
-def _do_publish(c):
+def _do_publish(client):
     """Internal function"""
-    m = c._userdata.pop()
-    if type(m) is dict:
-        topic = m['topic']
-        try:
-            payload = m['payload']
-        except KeyError:
-            payload = None
-        try:
-            qos = m['qos']
-        except KeyError:
-            qos = 0
-        try:
-            retain = m['retain']
-        except KeyError:
-            retain = False
-    elif type(m) is tuple:
-        (topic, payload, qos, retain) = m
+
+    message = client._userdata.pop()
+
+    if isinstance(message, dict):
+        client.publish(message['topic'],
```

Comment:
@Nzbuu,

Can I do the same here?

client.publish(**message)

If the dictionary has more keys than the function has parameters, we will get an error like this:

TypeError: publish() got an unexpected keyword argument 'foo'